stage1/init/pod.go

```diff
@@ -486,7 +487,14 @@ func (p *Pod) appToNspawnArgs(ra *schema.RuntimeApp) ([]string, error) {
 	for _, v := range p.Manifest.Volumes {
 		vols[v.Name] = v
 		if v.Kind == "empty" {
-			if err := os.MkdirAll(filepath.Join(common.SharedVolumesPath(p.Root), v.Name.String()), 0755); err != nil {
+			sharedVolPath := common.SharedVolumesPath(p.Root)
+			if err := os.MkdirAll(sharedVolPath, sharedVolPerm); err != nil {
+				return nil, fmt.Errorf("could not create shared volumes directory: %v", err)
+			}
+			if err := os.Chmod(sharedVolPath, sharedVolPerm); err != nil {
+				return nil, fmt.Errorf("could not change permissions of %q: %v", sharedVolPath, err)
+			}
```

Comment:
I'll move the first `MkdirAll()` and the `Chmod()` before the for loop so we ensure `/sharedVolumes` doesn't have `setgid`. Then, we I create the empty volume inside the loop it will also not have the `setgid`.